pkg/types/types.go

```diff
@@ -197,29 +197,57 @@ type Metric struct {
 }
 
 // MergeMetrics merges metrics by name.
-func MergeMetrics(metrics [][]Metric) []Metric {
+func MergeMetrics(metrics [][]Metric, consistencyCheck bool) ([]Metric, int, int) {
 	if len(metrics) == 0 {
-		return nil
+		return nil, 0, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		pointCount := 0
+		ms := metrics[0]
+		for _, m := range ms {
+			pointCount += len(m.Values)
+		}
+		return ms, pointCount, 0
 	}
 
-	names := make(map[string][]Metric)
+	metricByNames := make(map[string][]Metric)
 
 	for _, ms := range metrics {
 		for _, m := range ms {
-			names[m.Name] = append(names[m.Name], m)
+			metricByNames[m.Name] = append(metricByNames[m.Name], m)
 		}
 	}
 
 	merged := make([]Metric, 0)
-	for _, ms := range names {
-		merged = append(merged, mergeMetrics(ms))
+	pointCount := 0
+	inconsistencyCount := 0
+	InconsistentMetricsLogLimit := 10
+	var limitedInconsistentMetricsForLog []map[string]interface{}
+	for _, ms := range metricByNames {
+		m, c := mergeMetrics(ms, consistencyCheck)
+		if c > 0 && len(limitedInconsistentMetricsForLog) < InconsistentMetricsLogLimit {
+			limitedInconsistentMetricsForLog = append(limitedInconsistentMetricsForLog, map[string]interface{}{
+				"metricName":         m.Name,
+				"start":              m.StartTime,
+				"stop":               m.StopTime,
+				"step":               m.StepTime,
+				"inconsistentPoints": c,
+			})
+		}
+		merged = append(merged, m)
+		inconsistencyCount += c
+		pointCount += len(m.Values)
 	}
 
-	return merged
+	if consistencyCheck && inconsistencyCount > 0 {
+		corruptionLogger.Warn("metric consistency issue",
+			zap.Any("inconsistentMetric", limitedInconsistentMetricsForLog),
```

Comment:
Inconsistency seems to generic a term to understand what's the actual issue. Maybe something like replicaMismatch would be more clear? Other ideas?

Reply:
Well I agree with you that it is a general term but if you look at it in the context of `Render` and `Merge`, I guess it does the job. But if you don't feel the same, we can consider changing it to something like `replicaMismatch` as you proposed.

Reply:
@bom-d-van wdyt?

Reply:
`replicaMismatch` is indeed more descriptive. for some reason, we just keep using inconsistent/inconsistency for the whole issue. if we changed it here, we might have to change all other names with `inconsistent` to `replicaMismatch` in the PR? If so, maybe it's easier for us to stick with `inconsistent`?

Reply:
If we agree that replicaMismatch is better, it's best to change it once rather than keep less descriptive wording forever.

Reply:
please check @timtofan.